### PR TITLE
DAOS-3031 test: Improve node_local_test and resolve failures.

### PR DESCRIPTION
**Pull request by ashleypittman**
Improve the node_local_test script:
Do not save logs from previous runs
Shutdown the IO server directly
Check server log, but do not include memory leaks
Set master as a reference build.
Whitelist a number of allocation mismatches.

Fix some memory leaks.
Corret an issue in dfuse where UNS hash table entry was
incorrect, leading to a use-after-free and double-free
issue.

### CODE REVIEW

File: utils/node_local_test.py [hunk 1]
```diff
@@ -52,12 +52,14 @@ def __init__(self, conf):
         self.conf = conf
         self._agent = None
         self.agent_dir = None
+        # Also specified in the yaml file.
+        self._log_file = '/tmp/dnt_server.log'
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply utils/node_local_test.py:56]: Fixed filenames in /tmp are a security risk so my plan is to use tempfile() for this, and then create the yaml file on-the-fly.

File: src/client/dfuse/ops/lookup.c [hunk 2]
```diff
@@ -256,7 +256,7 @@ check_for_uns_ep(struct dfuse_projection_info *fs_handle,
 	if (rc)
 		D_GOTO(out_umount, ret = rc);
 
-	rc = dfuse_lookup_inode(fs_handle, dfs, &oid,
+	rc = dfuse_lookup_inode(fs_handle, dfs, NULL,
```
Comment (jolivier23) [inline src/client/dfuse/ops/lookup.c:259]: What does this do?

Comment (ashleypittman) [reply src/client/dfuse/ops/lookup.c:259]: This is a bug-fix to dfuse, if a path is accessed via the container path, and also through an unified namespace entry-point it's entered into the hash table with two different sets of options, so is marked as two different inodes, which can then lead to access-after-free or similar during forget calls from the kernel.

File: utils/node_local_test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (brianjmurrell) [reply utils/node_local_test.py:56]: In this PR?  Just wondering if you've forgotten about this or are deferring it?

[the remark this answers is not in the document]
Comment (ashleypittman) [reply utils/node_local_test.py:56]: No, I want to get this feature enabled first.  Not a regression, it's just the correct fix is out-of-scope for this PR.

**Pull request merged**